src/components/Camera/ARCamera/ARCamera.js

```diff
@@ -111,11 +113,15 @@ const ARCamera = ( {
     return 5;
   };
 
-  useEffect( ( ) => {
+  useEffect( () => {
+    if ( hasFinishedHere ) {
+      return;
+    }
     if ( photoSaved ) {
+      setHasFinishedHere( true );
       navToObsEdit( localTaxon );
     }
-  }, [photoSaved, navToObsEdit, localTaxon] );
+  }, [hasFinishedHere, photoSaved, navToObsEdit, localTaxon] );
```

Comment:
What is this for? This either needs a better variable name or a comment explaining why `hasFinishedHere` is necessary.

Reply:
Have renamed and added a comment.